packages/react-reconciler/src/ReactChildFiber.new.js

```diff
@@ -293,6 +302,14 @@ function ChildReconciler(shouldTrackSideEffects) {
       returnFiber.deletions = [childToDelete];
       // TODO (effects) Rename this to better reflect its new usage (e.g. ChildDeletions)
       returnFiber.effectTag |= Deletion;
+
+      // If we are deleting a subtree that contains a passive effect,
+      // mark the parent so that we're sure to traverse after commit and run any unmount functions.
+      const primaryEffectTag = childToDelete.effectTag & PassiveMask;
+      const primarySubtreeTag = childToDelete.subtreeTag & PassiveSubtreeTag;
+      if (primaryEffectTag !== NoEffect || primarySubtreeTag !== NoSubtreeTag) {
```

Comment:
Maybe this optimization isn't worth it and we  should always set the PassiveSubtreeTag on the parent when a child is deleted. i.e. add Deletion to PassiveMask.

Effects are very common — any large tree is almost certain to have at least one. So the optimization really only kicks in when you delete a leaf node. Not sure it's worth the extra code.

I'm almost tempted to say that PassiveStatic isn't worth it for the same reason; but since we're planning to have other static tags for things like portals, PassiveStatic doesn't add any net additional overhead, so might as well keep it.

What do you think?

Reply:
> Maybe this optimization isn't worth it and we should always set the PassiveSubtreeTag on the parent when a child is deleted.

Given that the concept of static flags seems like a positive– and so keeping `PassiveStatic` around doesn't add any net overhead– the only question seems to be whether the few extra bytes required for us to check `effectTag` and `subtreeTag` are worth it, yeah?

With the current check, we might over traverse in the event that a child was deleted that had a passive effect, but no cleanup function. (Seems unlikely.) If we removed this check, we might over traverse any time a leaf node is deleted, period. Still, I guess that wouldn't be *that bad* since we'd only traverse the path to where the leaf was deleted.

TBH I think the cost is pretty small on either side (number or bytes or potential over traversing). Happy to remove the check for now.

Reply:
> If we removed this check, we might over traverse any time a leaf node is deleted, period. Still, I guess that wouldn't be that bad since we'd only traverse the path to where the leaf was deleted.

Yeah I think that's fine. It's also related to the `detachFiber` discussion. Since we're moving most of that work to the passive phase, then we need to visit every deletion during the passive phase anyway.

Reply:
That's a good point.

Reply:
That's a good point I guess. I hadn't considered that aspect.

I consider the "detach" methods as kind of an optional optimization.